lib/puppet/util/windows/adsi.rb

```diff
@@ -464,12 +464,7 @@ def remove_member_sids(*sids)
     end
 
     def members
-      # WIN32OLE objects aren't enumerable, so no map
-      members = []
-      # Setting WIN32OLE.codepage in the microsoft_windows feature ensures
-      # values are returned as UTF-8
-      native_group.Members.each {|m| members << m.Name}
-      members
+      member_sids.map(&:domain_account)
     end
 
     def member_sids
```

Comment:
Not sure yet if this should be memoized like

```ruby
@member_sids ||= self.class.get_sids(native_group.Members)
```

Reply:
May not be worth it.  The class doesn't live that long.